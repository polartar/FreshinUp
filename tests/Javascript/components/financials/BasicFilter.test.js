import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/financials/BasicFilter.vue'

describe('BasicFilter', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('select functions change filters', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          filters: {
            event_uuid: null,
            host_uuid: null,
            store_uuid: null,
            supplier_uuid: null,
            date_after: null,
            date_before: null
          }
        }
      })
      wrapper.vm.selectEvent({ uuid: 1 })
      expect(wrapper.props().filters.event_uuid).toBe(1)
      wrapper.vm.selectHost({ uuid: 2 })
      expect(wrapper.props().filters.host_uuid).toBe(2)
      wrapper.vm.selectStore({ uuid: 3 })
      expect(wrapper.props().filters.store_uuid).toBe(3)
      wrapper.vm.selectSupplier({ uuid: 4 })
      expect(wrapper.props().filters.supplier_uuid).toBe(4)
    })
    test('changeDate() set values to date_after and date_before filters', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          filters: {
            event_uuid: null,
            host_uuid: null,
            store_uuid: null,
            supplier_uuid: null,
            date_after: null,
            date_before: null
          }
        }
      })
      wrapper.setData({ range: {
        start: '2019-12-11',
        end: '2019-12-17'
      } })
      wrapper.vm.changeDate()
      expect(wrapper.props().filters.date_after).toBe('2019-12-11')
      expect(wrapper.props().filters.date_before).toBe('2019-12-17')
    })
  })
  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('searchLink', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          filters: {
            event_uuid: 1,
            host_uuid: 2,
            store_uuid: 3,
            supplier_uuid: 4,
            date_after: '2019-12-11',
            date_before: '2019-12-17'
          }
        }
      })
      expect(wrapper.vm.searchLink).toBe('/admin/financials/transactions?event_uuid=1&host_uuid=2&store_uuid=3&supplier_uuid=4&date_after=2019-12-11&date_before=2019-12-17')
    })
    test('storeUrl with supplier_uuid filter not set', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          filters: {
            event_uuid: null,
            host_uuid: null,
            store_uuid: null,
            supplier_uuid: null,
            date_after: null,
            date_before: null
          }
        }
      })
      expect(wrapper.vm.storeUrl).toBe('foodfleet/stores')
    })
    test('storeUrl with supplier_uuid filter set', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          filters: {
            event_uuid: null,
            host_uuid: null,
            store_uuid: null,
            supplier_uuid: 1,
            date_after: null,
            date_before: null
          }
        }
      })
      expect(wrapper.vm.storeUrl).toBe('foodfleet/stores?filter[supplier_uuid]=1')
    })
  })
  describe('data()', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('range is set if date_before and/or date_after are set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          filters: {
            event_uuid: 1,
            host_uuid: 2,
            store_uuid: 3,
            supplier_uuid: 4,
            date_after: '2019-12-11',
            date_before: '2019-12-17'
          }
        }
      })
      expect(wrapper.vm.range).toEqual({
        start: '2019-12-11',
        end: '2019-12-17'
      })
    })
  })
})
