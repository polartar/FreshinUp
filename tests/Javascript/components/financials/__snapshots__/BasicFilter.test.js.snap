// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicFilter Snapshots defaults 1`] = `
<div
  class="layout row wrap"
>
  <div
    class="flex sm2"
  >
    <div
      class="layout row justify-space-between mb-2"
    >
      
      Events
      
      <!---->
    </div>
     
    <div
      background-color="white"
      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      height="48"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot white"
          style="height: 48px;"
        >
          <div
            class="v-select__slot"
          >
            <input
              aria-label=""
              autocomplete="off"
              placeholder="All Events"
              role="combobox"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex sm2 px-2"
  >
    <div
      class="layout row justify-space-between mb-2"
    >
      
      Date range
      
      <!---->
    </div>
     
    <div
      class="date-time-picker"
      id="DateTimePicker"
    >
      <div
        class="field flex align-center lg"
        data-v-a0144ed0=""
      >
        <input
          class="field-input"
          data-v-a0144ed0=""
          id="DateTimePicker"
          placeholder="Select date"
          readonly="readonly"
          type="text"
        />
        <label
          class="field-label"
          data-v-a0144ed0=""
          for="DateTimePicker"
        >
          
    Select date
  
        </label>
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    class="flex sm2 px-2"
  >
    <div
      class="layout row justify-space-between mb-2"
    >
      
      Companies
      
      <!---->
    </div>
     
    <div
      background-color="white"
      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      height="48"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot white"
          style="height: 48px;"
        >
          <div
            class="v-select__slot"
          >
            <input
              aria-label=""
              autocomplete="off"
              placeholder="All Companies"
              role="combobox"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex sm2 px-2"
  >
    <div
      class="layout row justify-space-between mb-2"
    >
      
      Truck
      
      <!---->
    </div>
     
    <div
      background-color="white"
      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      height="48"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot white"
          style="height: 48px;"
        >
          <div
            class="v-select__slot"
          >
            <input
              aria-label=""
              autocomplete="off"
              placeholder="All Trucks"
              role="combobox"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex sm2 px-2"
  >
    <div
      class="layout row justify-space-between mb-2"
    >
      
      Customer
      
      <!---->
    </div>
     
    <div
      background-color="white"
      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
      height="48"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot white"
          style="height: 48px;"
        >
          <div
            class="v-select__slot"
          >
            <input
              aria-label=""
              autocomplete="off"
              placeholder="All Customers"
              role="combobox"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--"
              >
                <i
                  aria-hidden="true"
                  class="v-icon v-icon--link material-icons theme--light"
                />
              </div>
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  arrow_drop_down
                </i>
              </div>
            </div>
          </div>
          <div
            class="v-menu"
          >
            <div
              class="v-menu__content theme--light v-autocomplete__content"
              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
            >
              <div
                class="v-select-list v-card theme--light"
              >
                <div
                  class="v-list theme--light"
                  role="list"
                >
                  <div
                    role="listitem"
                  >
                    <div
                      class="v-list__tile theme--light"
                    >
                      <div
                        class="v-list__tile__content"
                      >
                        <div
                          class="v-list__tile__title"
                        >
                          No data available
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex sm2 pt-4 fill-height justify-end pl-2"
  >
    <a
      class="primary v-btn v-btn--large theme--light"
      href="/admin/searches?"
    >
      <div
        class="v-btn__content"
      >
        
      Generate
    
      </div>
    </a>
  </div>
</div>
`;
